=============================================================================
Nama    : Rendi Dwi Yulian

Objective Umum:

        Program ini dibuat untuk dapat memperkirakan seberapa besar potensi keuntungan yang bisa diperoleh dari berjualan seblak dengan skema dropship. Untuk itu diperlukan data penjualan seblak yang diperoleh dari melakukan Web scraping pada marketplace Tokopedia. lalu akan dilakukan data cleaning sebelum melakukan eksplorasi dan analisa lebih jauh. kemudian dilakukan dilakukan beberapa pengujian statistik untuk mendapatkan data-data yang diinginkan.



=============================================================================

In [187]:
#library & modul
import pandas as pd
import requests
from bs4 import BeautifulSoup
from selenium import webdriver
import time
import csv
from scipy import stats
import numpy as np

In [215]:
Seblak=[]
#list kosong untuk menampung data Seblak yang akan di scraping

driver = webdriver.Edge() 
# digunakan untuk membuat instance dari WebDriver yang akan digunakan untuk mengontrol browser Microsoft Edge

for i in range(1,18): 
# perintah looping untuk link Tokped dari halaman 1-17
    Tokped="https://www.tokopedia.com/search?navsource=&page={}&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st=".format(i)
    #link Tokped 
    driver.get(Tokped) 
    # perintah untuk membuka halaman web dengan URL yang disediakan dalam variabel Tokped menggunakan WebDriver
    time.sleep(3) 
    # perintah untuk memberi waktu jeda selama 3 detik sebelum melanjutkan. Waktu jeda ini memberikan waktu bagi halaman web untuk sepenuhnya dimuat sebelum melakukan interaksi selanjutnya
    html = driver.page_source 
    #perintah untuk mengambil seluruh kode HTML dari halaman web yang saat ini dibuka dalam browser WebDriver
    soup = BeautifulSoup(html, "html.parser") 
    #perintah untuk membuat objek BeautifulSoup yang akan digunakan untuk melakukan parsing dan ekstraksi data dari sumber kode HTML.
       

    for sblk in soup.findAll('div', class_="css-uwyh54")[:10]: 
        # perintah loop untuk melakukan iterasi pada elemen class 'css-uwyh54' dalam objek soup. Elemen ini adalah elemen yang berisi informasi produk pada halaman web. 
        # [10] perintah untuk hanya mengambil 10 elemen pertama dari hasil pencarian.
        namaproduk          = sblk.find('div', class_="css-3um8ox").text  
        # perintah untuk mencari elemen <div> class 'css-3um8ox' di dalam elemen sblk yang sedang diiterasi. Elemen ini berisi nama produk. 
        # (.text) untuk mengambil teks di dalam elemen tersebut dan menyimpannya dalam variabel namaproduk.
        hargaproduk         = sblk.find('div', class_="css-1ksb19c").text
        # perintah mencari elemen dengan class 'css-1ksb19c' yang berisi harga produk. fungsi (.text) sama dengan penjelasan sebelumnya(di atas)
        
        for sblk2 in sblk.findAll('div', class_="css-1rn0irl"): 
        # perintah loop untuk melakukan iterasi pada elemen class 'css-uwyh54' dalam objek soup.  
            seler           = sblk2.findAll('span', class_="css-1kdc32b")[1].text
            # perintah untuk mencari semua elemen <span> dengan class 'css-1kdc32b'. Elemen ini berisi informasi penjual. 
            # [0] menunjukkan untuk mengambil elemen pertama dari hasil pencarian (asumsi bahwa elemen pertama berisi nama penjual).  fungsi (.text) sama dengan penjelasan sebelumnya(di atas)
            city            = sblk2.findAll('span', class_="css-1kdc32b")[0].text
            # perintah untuk mencari semua elemen <span> dengan class 'css-1kdc32b'. Elemen ini berisi informasi nama kota. 
            # [0] menunjukkan untuk mengambil elemen pertama dari hasil pencarian (asumsi bahwa elemen pertama berisi nama penjual).  fungsi (.text) sama dengan penjelasan sebelumnya(di atas)
        # pada case ini, elemen nama penjual/nama toko dan asal kota berada didalam 1 class yang sama, yaitu css-1rn0irl. 
        # maka perlu dilakukan looping untuk melakukan scaping untuk mendapatkan informasi yang diinginkan satu per satu

        terjual = sblk.findAll('span', class_="css-1duhs3e")
        # perintah untuk mencari semua elemen <span> dengan class 'css-1duhs3e' di dalam elemen sblk yang sedang diiterasi. 
        # Elemen ini berisi informasi tentang total penjualan produk.
        if len(terjual)>0:
        # kondisi yang memeriksa apakah hasil pencarian terjual menghasilkan setidaknya satu elemen. 
        # jika jumlah elemen dari terjual lebih besar dari 0, berarti informasi total penjualan ditemukan.
            totaljual       = sblk.find('span', class_="css-1duhs3e").text
            # perintah mencari elemen <span> dengan class 'css-1duhs3e' di dalam elemen sblk. Elemen ini berisi total penjualan.
            # fungsi (.text) sama dengan penjelasan sebelumnya(di atas)
        else: 
        # kondisi jika elemen tidak ditemukan
            totaljual       = None 
            # variabel totaljual akan diatur menjadi None jika total penjualan tidak tersedia atau tidak ditemukan.
        rating = sblk.findAll('span', class_="css-t70v7i")
        # perintah untuk mencari semua elemen <span> dengan class 'css-t70v7i' di dalam elemen sblk yang sedang diiterasi. 
        # Elemen ini berisi informasi tentang rating produk.
        if len(rating)>0:
        # kondisi yang memeriksa apakah hasil pencarian rating menghasilkan setidaknya satu elemen. 
        # jika jumlah elemen dari rating lebih besar dari 0, berarti informasi rating produk ditemukan.
            ratingproduk    = sblk.find('span', class_="css-t70v7i").text
            # perintah mencari elemen <span> dengan class 'css-t70v7i' di dalam elemen sblk. Elemen ini berisi rating penjualan.
            # fungsi (.text) sama dengan penjelasan sebelumnya(di atas)
        else:
        # kondisi jika elemen tidak ditemukan
            ratingproduk    = None
            # variabel totaljual akan diatur menjadi None jika total penjualan tidak tersedia atau tidak ditemukan.

        Seblak.append(
            (namaproduk, hargaproduk, seler, city, totaljual, ratingproduk)
            )
        # perintah untuk menambahkan namaproduk, hargaproduk, seler, city, totaljual, ratingproduk  yang diperoleh di atas
        # ke dalam list kosong Seblak yang sudah dibuat sebelumnya. 
        # Dengan melakukan ini, setiap kali informasi produk diambil dari halaman web, maka akan ditambahkan ke dalam list kosong Seblak.

df_Seblak=pd.DataFrame(Seblak, columns=['Nama Produk', 'Harga Produk', 'Seler', 'Lokasi', 'Total Terjual', 'Rating'])
df_Seblak.head()

The msedgedriver version (115.0.1901.200) detected in PATH at c:\Users\Kuro\Desktop\FTDS Hacktiv8\Latihan\Phase0\GC 3\p0-ftds001-bsd-g3-rendidy\msedgedriver.exe might not be compatible with the detected edge version (115.0.1901.203); currently, msedgedriver 115.0.1901.203 is recommended for edge 115.*, so it is advised to delete the driver in PATH and retry


Nama Produk Harga Produk  \
0  Seblak Instan Ceu Nthien Khas Bandung Rasana N...     Rp15.000   
1  SEBLAK JUARA INSTAN MASAK BASAH ASLI BANDUNG E...     Rp22.000   
2                  KERUPUK SEBLAK MENTAH ANEKA WARNA     Rp11.500   
3  MASJAY Seblak Bumbu Membara Instant - No.1 Str...     Rp15.000   
4         Seblak Juara ! Seblak Instan Rasa Original    Rp115.000   

                      Seler             Lokasi  Total Terjual Rating  
0  Central Seblak Nusantara  Tangerang Selatan  Terjual 1 rb+    4.9  
1      Rasa Juara Indonesia            Bandung   Terjual 750+    4.9  
2            jajanangarut19    Jakarta Selatan   Terjual 100+    4.9  
3                  Lidigeli         Kab. Garut    Terjual 70+    5.0  
4      Rasa Juara Indonesia            Bandung  Terjual 2 rb+    4.9

### Penjelasan
Data di atas diperoleh dengan melakukan web scraping dari website tokopedia.
Terdapat Lebih dari 80ribu hasil pencarian "Seblak" yang ada di tokopedia. Untuk itu, saya menscraping menggunakan metode data random sampling dari hasil pencarian "Seblak" dengan dengan ketentuan mengambil 10 data per halaman, dan di ambil dari total 17 halaman. maka diperoleh scraping data hasil pencarian "Seblak" dari tokopedia sebanyak 170 baris data.

In [254]:
df = df_Seblak.to_csv('Seblak', index=False)

Mengkonversi dataframe menjadi file CSV. agar data yang diperoleh bisa konsisten dan tidak berubah.

In [255]:
Data_Seblak = pd.read_csv('Seblak')

Memuat file CSV yang dikonvert sebelumnya, untuk di olah lebih lanjut

In [256]:
Data_Seblak
# menampilkan dataframe 

Nama Produk Harga Produk  \
0    Seblak Instan Ceu Nthien Khas Bandung Rasana N...     Rp15.000   
1    SEBLAK JUARA INSTAN MASAK BASAH ASLI BANDUNG E...     Rp22.000   
2                    KERUPUK SEBLAK MENTAH ANEKA WARNA     Rp11.500   
3    MASJAY Seblak Bumbu Membara Instant - No.1 Str...     Rp15.000   
4           Seblak Juara ! Seblak Instan Rasa Original    Rp115.000   
..                                                 ...          ...   
165                        Seblak Putih Odjolali Snack     Rp11.000   
166    Kerucang - Kerupuk Bumbu Kacang - 150 gr Seblak     Rp12.500   
167        Kerupuk Terasi Mentah 250gr /Kerupuk Seblak      Rp5.350   
168  PAKET 1 BOCI EXTRA PEDAS + 1 TULANG RANGU + 1 ...     Rp50.000   
169  SIOMAY MINI KERING 5 KG Garut campuran bahan b...    Rp140.999   

                        Seler             Lokasi  Total Terjual  Rating  
0    Central Seblak Nusantara  Tangerang Selatan  Terjual 1 rb+     4.9  
1        Rasa Juara Indonesia            Bandung   Terjual 750+     4.9  
2              jajanangarut19    Jakarta Selatan   Terjual 100+     4.9  
3                    Lidigeli         Kab. Garut    Terjual 70+     5.0  
4        Rasa Juara Indonesia            Bandung  Terjual 2 rb+     4.9  
..                        ...                ...            ...     ...  
165            Odjolali Snack            Bandung   Terjual 100+     5.0  
166                 cemilan4u            Bandung            NaN     NaN  
167                  Sin Nana           Semarang            NaN     NaN  
168                 JM Snacks            Bandung    Terjual 30+     4.9  
169            ArfaOnlineShop          Tangerang     Terjual 11     4.7  

[170 rows x 6 columns]

In [257]:
Data_Seblak['Nama Produk'].nunique()

150

Terdapat 150 data unique dari kolom nama produk dari data frame data_seblak di atas. itu berarti ada 20 duplicate data yang harus di bersihkan

In [259]:
Data_Seblak['Nama Produk'].value_counts()
#menghitung total masing-masing value dari kolom nama prduk untuk melihat data duplikat

Paket Adha 4 - Dimsum; Seblak Kering; Panzerotti; Seblak Komplit Kuah    3
Seblak Instan Ceu Nthien Khas Bandung Rasana Ngeunah Pisan               2
SEBLAK JUARA INSTAN MASAK BASAH ASLI BANDUNG ENAK MANTAP                 2
Keripik Seblak Pedas / Snack Spicy Homemade, Cemilan, Kerupuk, Jajan     2
Seblak Tulang Rawan k.y.l Kylafood                                       2
                                                                        ..
Seblak Pedas Kicimpring Iyagasih                                         1
Kerupuk Seblak Pedas 200gram                                             1
Seblak Putri 80 gr                                                       1
aneka kerupuk mix untuk bahan seblak 500 gr                              1
SIOMAY MINI KERING 5 KG Garut campuran bahan baso Aci dan seblak         1
Name: Nama Produk, Length: 150, dtype: int64

In [260]:
Data_Seblak.info()
#perintah untuk menampilkan detail info dari data frame

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 170 entries, 0 to 169
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Nama Produk    170 non-null    object 
 1   Harga Produk   170 non-null    object 
 2   Seler          170 non-null    object 
 3   Lokasi         170 non-null    object 
 4   Total Terjual  140 non-null    object 
 5   Rating         132 non-null    float64
dtypes: float64(1), object(5)
memory usage: 8.1+ KB


Analisa:

    1. Terlihat pada detail info di atas, tipe data dari masing-masing kolom, ada yang belum sesuai dengan tipe data yang seharusanya. 
    2. Terdapat juga beberapa Standart Missing value yang terdeteksi.
    3. Jika diperhatikan lebih lanjut, juga terdapat data duplikat yang terscraping dari beberapa halaman.
    
hal ini menunjukan, bahwa diperlukan data cleaning terlebih dahulu sebelum data di analisa lebih lanjut.


In [263]:
#Memperbaiki Data dari kolom, dengan menghapus / mengubah value yang tidak sesuai aataupun tidak dibutuhkan

Data_Seblak['Harga Produk'] = Data_Seblak['Harga Produk'].str.replace('Rp', '') 
#fungsi untuk menghapus Rp (rupiah)
Data_Seblak['Harga Produk'] = Data_Seblak['Harga Produk'].str.replace('.', '') 
#fungsi untuk menghapus (.)
Data_Seblak['Total Terjual'] = Data_Seblak['Total Terjual'].str.replace(' ', '') 
#fungsi untuk menghapus ' '(spasi)
Data_Seblak['Total Terjual'] = Data_Seblak['Total Terjual'].str.replace('+', '') 
#fungsi untuk menghapus simbol ('+')
Data_Seblak['Total Terjual'] = Data_Seblak['Total Terjual'].str.replace('Terjual', '') 
#fungsi untuk menghapus tulisan 'terjual'
Data_Seblak['Total Terjual'] = Data_Seblak['Total Terjual'].str.replace('rb', '000')
#fungsi untuk mengubah 'rb' menjadi '000'


C:\Users\Kuro\AppData\Local\Temp\ipykernel_10540\2196735131.py:5: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  Data_Seblak['Harga Produk'] = Data_Seblak['Harga Produk'].str.replace('.', '')
C:\Users\Kuro\AppData\Local\Temp\ipykernel_10540\2196735131.py:9: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  Data_Seblak['Total Terjual'] = Data_Seblak['Total Terjual'].str.replace('+', '')


- menghilangkan elemen-elemen yang tidak dipelukan di dalam data, sebelum dilakukan perubahan tipe data yang sesuai.
- merubah 'rb' menjadi satuan angka '000' 

In [264]:
# Mengganti nilai 'None' dengan 0 pada kolom 'Total Terjual' dan 'Rating'
Data_Seblak['Total Terjual'] = Data_Seblak['Total Terjual'].fillna(0)
Data_Seblak['Rating'] = Data_Seblak['Rating'].fillna(0)

- pada kolom total terjual dan rating, terdapat missing value berupa 'None' yang menandakan bahwa produk tersebut belum ada yang terjual ataupun belum ada yang memberi rating.

In [266]:
# Merubah Tipe Data kolom

Data_Seblak['Harga Produk'] = Data_Seblak['Harga Produk'].astype(int) 
#fungsi untuk merubah data(Object) Harga Produk menjadi data(integer)
Data_Seblak['Total Terjual'] = Data_Seblak['Total Terjual'].astype(int) 
#fungsi untuk merubah data(Object) Total Terjual menjadi data(integer)
Data_Seblak['Rating'] = Data_Seblak['Rating'].astype(float) 
#fungsi untuk merubah data(Object) Rating menjadi data(Float)

- merubah tipe data dari kolom pada data_seblak, agar data dapat di olah menjadi lebih lanjut.

In [267]:
Data_Seblak.duplicated().sum()
#perintah menghitung jumlah duplikat

20

In [268]:
# perintah untuk menghilangkan data duplikat
df = Data_Seblak.drop_duplicates()
df

Nama Produk  Harga Produk  \
0    Seblak Instan Ceu Nthien Khas Bandung Rasana N...         15000   
1    SEBLAK JUARA INSTAN MASAK BASAH ASLI BANDUNG E...         22000   
2                    KERUPUK SEBLAK MENTAH ANEKA WARNA         11500   
3    MASJAY Seblak Bumbu Membara Instant - No.1 Str...         15000   
4           Seblak Juara ! Seblak Instan Rasa Original        115000   
..                                                 ...           ...   
165                        Seblak Putih Odjolali Snack         11000   
166    Kerucang - Kerupuk Bumbu Kacang - 150 gr Seblak         12500   
167        Kerupuk Terasi Mentah 250gr /Kerupuk Seblak          5350   
168  PAKET 1 BOCI EXTRA PEDAS + 1 TULANG RANGU + 1 ...         50000   
169  SIOMAY MINI KERING 5 KG Garut campuran bahan b...        140999   

                        Seler             Lokasi  Total Terjual  Rating  
0    Central Seblak Nusantara  Tangerang Selatan           1000     4.9  
1        Rasa Juara Indonesia            Bandung            750     4.9  
2              jajanangarut19    Jakarta Selatan            100     4.9  
3                    Lidigeli         Kab. Garut             70     5.0  
4        Rasa Juara Indonesia            Bandung           2000     4.9  
..                        ...                ...            ...     ...  
165            Odjolali Snack            Bandung            100     5.0  
166                 cemilan4u            Bandung              0     0.0  
167                  Sin Nana           Semarang              0     0.0  
168                 JM Snacks            Bandung             30     4.9  
169            ArfaOnlineShop          Tangerang             11     4.7  

[150 rows x 6 columns]

In [269]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 150 entries, 0 to 169
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Nama Produk    150 non-null    object 
 1   Harga Produk   150 non-null    int32  
 2   Seler          150 non-null    object 
 3   Lokasi         150 non-null    object 
 4   Total Terjual  150 non-null    int32  
 5   Rating         150 non-null    float64
dtypes: float64(1), int32(2), object(3)
memory usage: 7.0+ KB


- Dari detail info di atas, terlihat tipe data dari masing-masing kolom sudah sesuai dengan tipe data yang seharusnya.
- elemen-elemen yang tidak diperlukan juga sudah dihilangkan.
- dan total data setelah menghilangkan dupilkat data adalah 150 row.

### Business Understanding & Problem Statement

    
Berikut ini adalah SMART goal untuk memulai bisnis dropship seblak di Tokopedia:

- Specific: Menganalisa data penjualan seblak dan memulai bisnis dropship seblak di Tokopedia dalam waktu 1 bulan.
- Measurable: Mencapai target penjualan seblak sebesar Rp10 juta dalam waktu 3 bulan.
- Achievable: Dengan kerja keras dan strategi yang tepat, target penjualan ini dapat dicapai.
- Relevant: Bisnis dropship seblak sangat relevan dengan tren saat ini, yaitu masyarakat yang menyukai makanan pedas.
- Time-bound: Target penjualan Rp10 juta harus dicapai dalam waktu 3 bulan.


Memulai bisnis dropship Seblak di Tokopedia dalam waktu 1 bulan, dan mampu meraih target penjualan sebesar 10jt dalam waktu 3 bulan dengan menerapkan strategi yang tepat.


## Analisa Statistik
- Selanjutnya akan dilakukan analisa statistik dari dataframe diatas, dengan melakukan penghitungan mean, median, standard deviasi, skew, dan kurtosis.

## 1.

In [272]:
meanharga = df["Harga Produk"].mean()
medianharga = df["Harga Produk"].median()
sdeviasiharga = df["Harga Produk"].std()
skewharga = df["Harga Produk"].skew()
kurtoharga = df["Harga Produk"].kurtosis()

print("Mean Harga Produk : ", meanharga)
print("Median Harga Produk : ", medianharga)
print("Standard Deviasi Harga Produk : ", sdeviasiharga)
print("Skewness Harga Produk : ", skewharga)
print("Kurtosis Produk : ", kurtoharga)


Mean Harga Produk :  24648.966666666667
Median Harga Produk :  15000.0
Standard Deviasi Harga Produk :  26873.464696065766
Skewness Harga Produk :  2.743929803982138
Kurtosis Produk :  9.44809243950047


- terlihat bahwa, harga rata-rata penjualan seblak adalah 24649 rupiah
- harga tengahnya adalah 15000 rupiah 
- dispersi harganya adalah 26873 rupiah
- dan terlihat skewness nya bernilai positif/ miring ke kanan.
- serta kurtosis nya bernilai positif yang berarti data akan lebih terpusat ke tengah.

In [273]:
meantj = df["Total Terjual"].mean()
mediantj = df["Total Terjual"].median()
stdtj = df["Total Terjual"].std()
skewtj = df["Total Terjual"].skew()
kurtj = df["Total Terjual"].kurtosis()

print("Mean Total Terjual : ", meantj)
print("Median Total Terjual : ", mediantj)
print("Standard Deviasi Total Terjual : ", stdtj)
print("Skewness Total Terjual : ", skewtj)
print("Kurtosis Produk : ", kurtj)

Mean Total Terjual :  361.2866666666667
Median Total Terjual :  40.0
Standard Deviasi Total Terjual :  1119.6704113364658
Skewness Total Terjual :  5.102407481262641
Kurtosis Produk :  29.521832509378136


- terlihat bahwa, penjualan rata-rata seblak adalah 361 pcs
- nilai tengah penjualan nya adalah 40 pcs  
- dispersi penjualannya adalah 1119 pcs
- dan terlihat skewness nya bernilai positif/ miring ke kanan.
- serta kurtosis nya bernilai positif yang berarti data akan lebih terpusat ke tengah.

In [274]:
meanrt = df["Rating"].mean()
medianrt = df["Rating"].median()
stdrt = df["Rating"].std()
skewrt = df["Rating"].skew()
kurt = df["Rating"].kurtosis()

print("Mean Rating : ", meanrt)
print("Median Rating : ", medianrt)
print("Standard Deviasi Rating : ", stdrt)
print("Skewness Rating : ", skewrt)
print("Kurtosis Produk : ", kurt)

Mean Rating :  3.874
Median Rating :  4.9
Standard Deviasi Rating :  1.987492096138942
Skewness Rating :  -1.4504481824689495
Kurtosis Produk :  0.12565785186835177


- terlihat bahwa, Rating rata-rata seblak adalah 3.8
- nilai tengah Rating nya adalah 4.9
- dispersi Ratingnya adalah 1.9
- dan terlihat skewness nya bernilai negatif/ miring ke kiri.
- serta kurtosis nya bernilai positif yang berarti data akan lebih terpusat ke tengah.

## 2.

- Setelah memperoleh data di atas, selanjutnya adalah memperkirakan berapa potensi keuntungan terkecil dan keuntungan terbesar yang mungkin bisa di peroleh dengan melakukan penjualan Seblak di tokopedia. 

In [275]:
std = df['Total Terjual'].std()
N = len(df)
low, up = stats.norm.interval(0.95,loc=df['Total Terjual'].mean(),scale=std/np.sqrt(N))
print('Lower Limit:',low)
print('Upper Limit:',up)

Lower Limit: 182.10537495150697
Upper Limit: 540.4679583818264


- lower limit menggambarkan potensi penjualan paling sedikit adalah 182 pcs
- upper limit menggambarkan potensi penjualan paling banyak adalah 540 pcs

In [276]:
Perkiraan_Maximal_Keuntungan = meanharga*up
Perkiraan_Maximal_Keuntungan

13321976.690555027

- Dengan pengalian antara Rata-rata harga x Potensi penjualan terbanyak, diperoleh potensi keuntungan terbesar yang bisa di dapat yaitu Rp 13321976, atau  13Juta+.

In [277]:
Perkiraan_Minimal_Keuntungan = meanharga*low
Perkiraan_Minimal_Keuntungan

4488709.317000531

- Dengan pengalian antara Rata-rata harga x Potensi penjualan paling sedikit, diperoleh potensi keuntungan terkecil yang bisa di dapat yaitu Rp 4488709, atau  4Juta+.

## 3.

- Setelah mendapatkan gambaran potensi keuntungan terkecil dan terbesar yang bisa diperoleh, selanjutnya adalah mempertimbangkan apakah harga produk seblak yang ada di jabodetabek dan luar jabodetabek. hal ini diperlukan karena ada faktor-faktor seperti harga bahan baku, biaya pengiriman dan lain lain yang dapat membantu dalam penentuan keputusan untuk melakukan dropship dari luar ataupun dalam jabodetabek.


In [278]:
df["Lokasi"].unique()
# perintah untuk menampilkan value unik yang menampilkan semua lokasi penjual dari data di atas

array(['Tangerang Selatan', 'Bandung', 'Jakarta Selatan', 'Kab. Garut',
       'Jakarta Barat', 'Kab. Bandung', 'Surabaya', 'Jakarta Timur',
       'Bekasi', 'Kab. Bogor', 'Jakarta Pusat', 'Tangerang', 'Medan',
       'Depok', 'Kab. Bantul', 'Sukabumi', 'Kab. Majalengka', 'Cimahi',
       'Kab. Tangerang', 'Kab. Tasikmalaya', 'Kab. Karawang',
       'Kab. Jember', 'Denpasar', 'Jakarta Utara', 'Tasikmalaya', 'Bogor',
       'Kab. Bekasi', 'Semarang', 'Cirebon', 'Kab. Badung'], dtype=object)

Terdapat 30 unique value dari nama lokasi penjual

In [280]:
# membuat list kota yang termasuk kedalam jabodetabek dari unique value di atas.
list_kota = ['Tangerang Selatan', 'Jakarta Selatan', 'Jakarta Barat', 
       'Jakarta Timur', 'Bekasi', 'Kab. Bogor', 'Jakarta Pusat', 
       'Tangerang', 'Depok', 'Kab. Tangerang', 'Jakarta Utara', 'Bogor', 'Kab. Bekasi']
#terdapat 13 kota yang termasuk kedalam area jabodetabek

# perintah untuk menampilkan data produk, jika lokasi nya ada di dalam list di atas  
Jabodetabek = df[df['Lokasi'].isin(list_kota)]

Jabodetabek.head()

Nama Produk  Harga Produk  \
0  Seblak Instan Ceu Nthien Khas Bandung Rasana N...         15000   
2                  KERUPUK SEBLAK MENTAH ANEKA WARNA         11500   
5  Seblak ala Kang Rafael | Seblak Viral | Seblak...          9216   
6  Kerupuk Seblak Viral / Kerupuk Mawar Putih Vir...          8000   
7                      Seblak Instan Pedas Home Made          3500   

                      Seler             Lokasi  Total Terjual  Rating  
0  Central Seblak Nusantara  Tangerang Selatan           1000     4.9  
2            jajanangarut19    Jakarta Selatan            100     4.9  
5       vieka mart official  Tangerang Selatan            250     5.0  
6         Bumbukeluargakita      Jakarta Barat             12     5.0  
7                the Dhecip  Tangerang Selatan           1000     4.8

In [281]:
# perintah untuk menampilkan data produk, jika lokasi nya tidak ada di dalam list kota  
Nonjabodetabek = df[~df['Lokasi'].isin(list_kota)]#menggunakan '~' untuk memilih selain yang berada di dalam list kota

Nonjabodetabek.head()

Nama Produk  Harga Produk  \
1   SEBLAK JUARA INSTAN MASAK BASAH ASLI BANDUNG E...         22000   
3   MASJAY Seblak Bumbu Membara Instant - No.1 Str...         15000   
4          Seblak Juara ! Seblak Instan Rasa Original        115000   
8              KERUPUK MENTAH SERBA 5000 BAHAN SEBLAK          6000   
10  Seblak Cocol Seblak Coet Seblak Viral Seblak B...          5000   

                   Seler        Lokasi  Total Terjual  Rating  
1   Rasa Juara Indonesia       Bandung            750     4.9  
3               Lidigeli    Kab. Garut             70     5.0  
4   Rasa Juara Indonesia       Bandung           2000     4.9  
8               GK_Snack  Kab. Bandung            500     5.0  
10        FAG COLLECTION      Surabaya              0     0.0

In [282]:
len(Jabodetabek)

80

Setelah dipisahkan antara data yang berasal dari dalam ataupun luar jabodetabek.
- Terdapat 80 data yang berasal dari dalam jabodetabek dan 70 data dari luar jabodetabek
Selanjunya membuat hipotesa untuk menentukan apakah harga berang di jabodetabek dan luar jabodetabek berbeda?

### Hipotesa:

- H0 : Harga Produk Seblak di Jabodetabek == Harga Produk Seblak diluar Jabodetabek.
- H1 : Harga Produk Seblak di Jabodetabek != Harga Produk Seblak diluar Jabodetabek.

In [249]:
t_stat, p_val = stats.ttest_ind(Jabodetabek['Harga Produk'],Nonjabodetabek['Harga Produk'])
print('T-Statistic:',t_stat)
print('P-value:',p_val) 

T-Statistic: 0.8539429970244128
P-value: 0.39451662993897374


Berdasarkan Nilai P value di atas, Terlihat bahwa  P-value (0.3945) lebih besar dari tingkat signifikansi umum (0.05).
- Hal ini menunjukan bahwa tidak memiliki cukup bukti untuk menolak hipotesis nol (H0). Perbedaan antara harga di jabodetabek dan diluar jabodetabek tidak signifikan.
- Maka H0 diterima


## 4

- Setelah mengetahui bahwa tidak ada perbedaan signifikan antara harga di jabodetabek dan diluar jabodetabek, selanjutnya harus mempertimbangkan apakan pembeli cendrung suka membeli harga produk seblak yang lebih murah dibandingkan yang lebih mahal

### Hipotesa:

- H0 : Tidak ada Korelasi antara Harga Produk Seblak Murah dengan Kecendrungan Pembelian customer
- H1 : Ada kecendrungan yang kuat antara Harga Produk Seblak yang lebih murah dengan minat pembelian customer 

In [250]:
corr_r, pval_p = stats.pearsonr(df['Harga Produk'], df['Total Terjual'])
print(f"r-correlation: {corr_r:.2f}, p-value: {pval_p}")

r-correlation: 0.04, p-value: 0.6187950993802246


- Terlihat dari r-correlation (0,04), menunjukan bahwa memiliki korelasi positif, namun juga menunjukan tidak ada korelasi yang kuat antara harga murah dengan kecendrungan minat pembeli.
- Maka, h0 diterima
- Tidak ada korelasi yang kuat antara harga murah dan minat pembeli

## Kesimpulan:

- Berdasarkan analisa statik di atas. Penjualan Seblak dengan cara Dropshiper melalui Tokopedia, Berpotensi menghasilkan keuntungan hingga 13 Juta+. Karenanya, Target 10jt penjualan dalam 3 bulan  dari Time bound masih mungkin untuk dicapai dengan strategi dan perencanaan yang baik.
- Harga produk Seblak di Jabodetabek dan diluar Jabodetabek tidaklah memiliki perbedaan yang signifikan, hal ini dapat digunakan untuk pertimbangan dalam memilih adal kota dropshiper dengan ongkir yang lebih murah untuk mendapatkan keuntungan yang lebih baik.
- Tidak ada korelasi antara harga jual yang murah dengan kecendrungan minat pembeli. hal ini masih bisa di ekplore lebih jauh, bisa jadi kecendrungan pembelian ini tergantung dengan rating ataupun review yang positif dari suatu produk.